### PR TITLE
Enhance ajd module

### PR DESCRIPTION
**Pull request by qbarthelemy**
This PR aims to:

- add a generic `ajd` function, solving partially #49;
- enhance functions `rjd` and `uwedge`, removing np.dot and np.diag(np.diag()), and using uniform notations;
- enhance `ajd_pham`, adding a warning when convergence is not reached; 
- ~~enhance `ajd_pham` to process HPD matrices, 
(following code https://github.com/Marco-Congedo/Diagonalizations.jl/blob/master/src/optim/LogLike.jl#L46). 
A consequence is that mean ALE supports HPD matrices (related to #204);~~
- complete tests.

### CODE REVIEW

File: pyriemann/utils/ajd.py [hunk 1]
```diff
@@ -231,84 +242,153 @@ def uwedge(X, *, init=None, eps=1e-7, n_iter_max=100):
...
+        C1 = np.zeros((n, n))
+        for i in range(n):
+            C1[:, i] = np.sum(Ms[:, i:n_matrices_x_n:n] * Rs, axis=1)
 
         D0 = B * B.T - np.outer(np.diag(B), np.diag(B))
-        A0 = (C1 * B - np.diag(np.diag(B)) @ C1.T) / (D0 + np.eye(d))
-        A0 += np.eye(d)
-        W_est = np.linalg.solve(A0, W_est)
+        A0 = (C1 * B - np.diag(np.diag(B)) @ C1.T) / (D0 + np.eye(n))
+        A0 += np.eye(n)
```
[the remark this answers is not in the document]
Comment (qbarthelemy) [reply pyriemann/utils/ajd.py:304]: Ok, I will replace
`np.diag(np.diag(B)) @ C1.T` by `np.diag(B)[:, np.newaxis] * C1.T`
`A0 += np.eye(n)` by `A0.flat[:: n + 1] += 1`
Is it ok for you?

File: pyriemann/utils/ajd.py [hunk 2]
```diff
@@ -231,84 +236,164 @@ def uwedge(X, *, init=None, eps=1e-7, n_iter_max=100):
...
         crit = crit_new
         iteration += 1
 
-    D = np.reshape(Ms, (d, n_matrices, d)).transpose(1, 0, 2)
-    return W_est, D
+    D = np.reshape(Ms, (n, n_matrices, n)).transpose(1, 0, 2)
+    return V, D
+
+
+###############################################################################
```
[the remark this answers is not in the document]
Comment (qbarthelemy) [reply pyriemann/utils/ajd.py:320]: This is only for developers, to separate different types of functions.

File: tests/test_utils_ajd.py [hunk 3]
```diff
@@ -3,69 +3,107 @@
...
 
+    V_, D_ = algo(mats, eps=eps, n_iter_max=n_iter_max)
+    assert np.all(V == V_)
+    assert np.all(D == D_)
 
-@pytest.mark.parametrize("ajd", [rjd, ajd_pham, uwedge])
-def test_ajd_init_error(ajd, get_covmats):
+
+@pytest.mark.parametrize("method", ["rjd", "ajd_pham", "uwedge"])
+@pytest.mark.parametrize("init", [True, False])
```
[the remark this answers is not in the document]
Comment (qbarthelemy) [reply tests/test_utils_ajd.py:40]: Yes, probably, but it gives the same result.

[the remark this answers is not in the document]
Comment (VascoSch92) [reply tests/test_utils_ajd.py:40]: yes but it is much better to read and to understand that you are taking al the pairs of the two lists. Or write directly the pairs

File: pyriemann/utils/ajd.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (qbarthelemy) [reply pyriemann/utils/ajd.py:320]: You can find these separations in most python packages.

Just an example
https://github.com/scikit-learn/scikit-learn/blob/5c4aa5d0d/sklearn/covariance/_shrunk_covariance.py#L99

**Review by agramfort — APPROVED** [review]

**Pull request merged**